test/e2e/network/ingressclass.go

```diff
@@ -83,18 +82,17 @@ var _ = common.SIGDescribe("IngressClass [Feature:Ingress]", func() {
 		defer deleteIngressClass(cs, ingressClass2.Name)
 
 		// the admission controller may take a few seconds to observe both ingress classes
-		expectedErr := "2 default IngressClasses were found, only 1 allowed"
 		var lastErr error
 		if err := wait.Poll(time.Second, time.Minute, func() (bool, error) {
 			defer cs.NetworkingV1().Ingresses(f.Namespace.Name).Delete(context.TODO(), "ingress1", metav1.DeleteOptions{})
-			_, err := createBasicIngress(cs, f.Namespace.Name)
+			ingress, err := createBasicIngress(cs, f.Namespace.Name)
```

Comment:
we have to rewrite this loop this way

```go
		if err := wait.Poll(time.Second, time.Minute, func() (bool, error) {
			_, err := createBasicIngress(cs, f.Namespace.Name)
			ingress, err := createBasicIngress(cs, f.Namespace.Name)
			if err != nil {
				return false, nil
			}
			if ingress.Name != ingressClass1.Name {
                                 return false, fmt.Errorf("expected ingress class %s but created with %s, ingressClass1.Name, ingress.Name)
                        }
                       return true, nil
		}); err != nil {
			framework.Failf("Failed to create ingress when two ingressClasses are marked as default ,got error %w", err)
		}
```

Reply:
Thanks for helping me find this bug 🤦‍♂️. Code updated.